### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException when using gradle plugin 3.0.0.x

### ISSUE THREAD

**Issue by gbourassa** [square/spoon#461]
When trying to use Spoon with the latest build tools / gradle plugin, the spooner-runner crashes when attempting to analyze the AndroidManifest of the test app:

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 96
	at com.squareup.spoon.internal.thirdparty.axmlparser.StringBlock.getShort(StringBlock.java:255)
	at com.squareup.spoon.internal.thirdparty.axmlparser.StringBlock.getRaw(StringBlock.java:110)
	at com.squareup.spoon.internal.thirdparty.axmlparser.StringBlock.read(StringBlock.java:77)
	at com.squareup.spoon.internal.thirdparty.axmlparser.AXMLParser.doStart(AXMLParser.java:201)
	at com.squareup.spoon.internal.thirdparty.axmlparser.AXMLParser.<init>(AXMLParser.java:48)
	at com.squareup.spoon.SpoonInstrumentationInfo.parseFromFile(SpoonInstrumentationInfo.java:62)
	at com.squareup.spoon.SpoonRunner.run(SpoonRunner.java:115)
	at com.squareup.spoon.Main.main(main.kt:35)
```

Maybe the AXML format has changed?

When using the Android Studio APK analyzer, the Manifest looks fine.

I reproduce using the master branch built locally. I added some log to try to figure out what is wrong. It's hard to say what the problem is. The length returned at line 95 of `StringBlock` (1285) seems way too big for my string blocks. `m_strings` contains only 96 entries. Hence the crash when attempting to get the 96th entry.

Thanks.

**Comment by duncancarroll** [comment]
I was able to get parsing to work using this lib: https://github.com/xgouchet/AXML
Some attribute names appear to have changed; not sure why the parsing was failing & didn't feel like debugging.

Quick and dirty gist if you just need to get unblocked: https://gist.github.com/duncancarroll/2a6307b3d21037dc73728702c95314bf

Can clean up later and PR if this is of interest to people.

**Comment by caipivara** [comment]
@gbourassa how did you get those logs? I'm having another issue but I cant find where it get broken

**Comment by duncancarroll** [comment]
This is output from the spoon-runner jar referenced in the main README under "Execution": https://search.maven.org/remote_content?g=com.squareup.spoon&a=spoon-runner&v=LATEST&c=jar-with-dependencies

**Comment by smuldr** [comment]
Could this issue be related to the UTF-8 / UTF-16 fix in PR #470?

**Comment by gbourassa** [comment]
@duncancarroll Sorry for the (very) late reply, I didn't get time recently to check on that.
I can confirm that using your code snippet and the AXML parser, it seems to work perfectly.
I'll try to cleanup you class and PR it, this could definitely be useful to some people.

**Comment by pavanagarwal** [comment]
Am seeing the same issue as mentioned above for my project. Is there any timeline for new gradle plugin release?

**Comment by edenman** [comment]
I've merged the PR that should fix the issue, but we don't have a timeline for when there will be another release of Spoon, so there's also no timeline for when the gradle plugin can be updated.  Sorry.

**Comment by mbmc** [comment]
For those running into this issue until a new release is available:
- `git clone git@github.com:square/spoon.git`
- `./gradlew installDebug`
- `export TEST_APK` to point to your androidTest.apk
- `export OTHER_APK` to point to your main.apk
- `java -jar spoon-runner-2.0.0-SNAPSHOT-all.jar $TEST_APK $OTHER_APK `

**Comment by handstandsam** [comment]
I've verified this is fixed in 2.0.0-SNAPSHOT if you build from source.  Note: A ton of command-line arguments have changed so see `CliArgs.kt` in the `spoon-runner` lib for details.

Also note: `installDebug` as mentioned above won't work for the `spoon-runner` lib.  You'll have to run `install` to get it to compile.

Are there any plans to expose the 2.0.0-SNAPSHOT publicly without having to build from source?

Thanks!

**Comment by JakeWharton** [comment]
The readme has a pointer to the snapshots Maven repo:
https://github.com/square/spoon/#download

**Comment by handstandsam** [comment]
Thanks @JakeWharton!

For those too lazy to browse to it: https://oss.sonatype.org/content/repositories/snapshots/com/squareup/spoon/spoon-runner/2.0.0-SNAPSHOT/

**Comment by JakeWharton** [comment]
Yeah the URL in the README is the one you'd add to your repositories{}
block in Gradle.

**Comment by sarmadali20** [comment]
Anyone tried AndroidTestOrchestrator with spoon?

**Comment by blazek** [comment]
The same problem here, I have downloaded https://oss.sonatype.org/content/repositories/snapshots/com/squareup/spoon/spoon-runner/2.0.0-SNAPSHOT/spoon-runner-2.0.0-20171020.002642-36-all.jar
but it is ignoring `-e KEY VALUE,  Instrumentation runner arguments.`
How can I set instrumentation options? I tried `-e key value` without success.

**Comment by gbourassa** [comment]
`--e` seems to work for me.

**Comment by blazek** [comment]
I found https://github.com/square/spoon/issues/490

**Comment by blazek** [comment]
> --e seems to work for me

It is working in spoon 1.7.x but not in 2.0 builds, there is an issue https://github.com/square/spoon/issues/490 and PR https://github.com/square/spoon/pull/495

**Comment by jaredsburrows** [comment]
You can close this now, this issue is fixed in master.

**Comment by chiprasa** [comment]
Could you please let me know when we can expect the final jar?

**Comment by sonnv8** [comment]
I run test spoon with the apk file built by gradle 2.2.3, then run normally, but if run with apk is built by gradle 3.0.1 it will fail "error com.squareup.spoon.axmlparser.StringBlock.getShort ( StringBlock.java:2 .... ". Help, please show me how to fix this error
